ext/tidy/tidy.c

```diff
@@ -1388,7 +1386,11 @@ PHP_METHOD(tidy, __construct)
 			RETURN_THROWS();
 		}
 
-		TIDY_APPLY_CONFIG(obj->ptdoc->doc, options_str, options_ht);
+		if (php_tidy_apply_config(obj->ptdoc->doc, options_str, options_ht) != SUCCESS) {
+			/* TODO: this is the constructor, we should throw probably... */
+			zend_string_release_ex(contents, 0);
+			RETURN_FALSE;
```

Comment:
not sure about this indeed.

Reply:
This keeps the existing behaviour of calling `->__construct(...)` manually, and yes it returns false in that case. Constructors don't necessarily have a void return :(
I believe this should be at least fixed for master by throwing an exception instead. cc @Girgias

Reply:
Yes this should, especially if the constructor can emit a warning